ismatec/driver.py

```diff
@@ -20,34 +31,36 @@ class Pump(Protocol):
     independently. See `self.channels` for available channels.
     """
 
-    def __init__(self, address=None, debug=False, **kwargs) -> None:
-        if debug:
-            logger.setLevel(logging.DEBUG)
-        """Initialize the Communicator and setup the pump to accept commands."""
-        # make a hardware Communicator object
-        if type(address) == str:
-            # serial
-            self.hw: Communicator = SerialCommunicator(address=address, **kwargs)
-        elif type(address) == tuple and len(address) == 2:
-            # socket
-            self.hw = SocketCommunicator(address=address, **kwargs)
+    def __init__(self, address='/dev/ttyUSB0', **kwargs) -> None:
+        """Connect to the pump and set initial state.
+
+        This optionally takes an address of the form
+            tcp://<<address>>:<<port>>
+        to talk to connected serial-to-ethernet converters.
+
+        Note that initialization sends multiple requests to the pump:
+        1. Enable independent channel addressing (TODO why?)
+        2. Disable asynchronous messages (TODO why?)
+        3. Get list of all available tube channels. (TODO why?)
+        4. Set initial running states (TODO why? to what?)
+        """
+        if address.startswith('tcp://'):
+            ip, port = address[6:].split(':')
+            self.hw: Communicator = SocketCommunicator(address=(ip, int(port)), **kwargs)
         else:
-            raise RuntimeError('Specify serial device or (host, port) tuple!')
-        self.hw.start()
+            self.hw = SerialCommunicator(address=address, **kwargs)
 
         # Enable independent channel addressing
-        self.hw.command('1~1')
+        self.hw.query('1~1')
```

Comment:
`command` and `query` should not be merged, since the pump will return different things over serial.  Commands get (mostly) `*` and `#` indicating success/failure, but queries get more data.